### DISCUSSION TITLE
Use Coredns as a custome dns infrastructure for forward query to different ip's

### DISCUSSION THREAD

**Discussion by PrakashRaj37** [coredns/coredns#4913] [Q&A] [answered]
How to use Coredns as a custome dns infrastructure for forward query to different ip's

**Comment by chrisohaver** [comment]
Please elaborate.

**Reply by PrakashRaj37** [reply]
We are developing docker container based application, Each user have separate container will run a service in port 3000 and each container should have unique url like service1.abc.example.org, service2.abd.example.org.
Our question is how to route specific url to specific container port. container will create dynamically how to do it with this package.

**Reply by chrisohaver** [reply]
Are you using, or do you plan to use Kubernetes?

**Reply by znyrraj** [reply]
No we are planning to develop our own container management system

**Comment by chrisohaver** [comment, marked as answer]
If your bespoke container management system will use etcd, you could perhaps use the _etcd_ plugin, but that expects records to be in a specific structure.

With a google search, I found https://github.com/kevinjqiu/coredns-dockerdiscovery. I am not familiar at all with the plugin, but maybe you could make use of it.

Ultimately though you are probably better off writing your own plugin that integrates with your bespoke container management system, as has been done for Kubernetes with the _kubernetes_ plugin.

**Reply by znyrraj** [reply]
Thanks for your reply @chrisohaver . We will go through those library incase it didn't suit for our need then we will write our own plugin.
Now we are facing some issues. We installed and run coredns in gcp vm instance and we bought our domain name for example "domain1.com" in bigrock. If anyone access "domain1.com" and all it subdomain should be managed by coredns which is installed in gcp instance. how to do it Thanks in advance.

**Reply by chrisohaver** [reply]
> Thanks for your reply @chrisohaver . We will go through those library incase it didn't suit for our need then we will write our own plugin.
> Now we are facing some issues. We installed and run coredns in gcp vm instance and we bought our domain name for example "domain1.com" in bigrock. If anyone access "domain1.com" and all it subdomain should be managed by coredns which is installed in gcp instance. how to do it Thanks in advance.

please ask this new question in a separate discussion.
